src/main/java/net/snowflake/ingest/streaming/internal/OpenChannelRequestInternal.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import net.snowflake.ingest.streaming.OpenChannelRequest;
+import net.snowflake.ingest.utils.Constants;
+import net.snowflake.ingest.utils.Utils;
+
+/** Class used to serialize the {@link OpenChannelRequest} */
+public class OpenChannelRequestInternal implements StreamingIngestRequest {
+  @JsonProperty("request_id")
+  private String requestId;
+
+  @JsonProperty("role")
+  private String role;
+
+  @JsonProperty("channel")
+  private String channel;
+
+  @JsonProperty("table")
+  private String table;
+
+  @JsonProperty("database")
+  private String database;
+
+  @JsonProperty("schema")
+  private String schema;
+
+  @JsonProperty("write_mode")
+  private String writeMode;
+
+  @JsonProperty("is_iceberg")
+  private boolean isIceberg;
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  @JsonProperty("offset_token")
+  String offsetToken;
+
+  OpenChannelRequestInternal(
+      String requestId,
+      String role,
+      OpenChannelRequest request,
```

Comment:
nit: lets just take all the channel/table/db/schema/offset properties as ctor arguments instead of passing in the OpenChannelRequest object. Preferable to keep public contract and internal contract as insulated from each other as possible.

Reply:
Exposed `database`, `schema`, `table`, `channel` as function signature.